compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -1407,6 +1419,27 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
             }
         }
 
+        // Aggregate-then-Transmute can just transmute the original field value,
+        // so long as the bytes of a value from only from a single field.
+        if let Transmute = kind
+            && let Value::Aggregate(
+                AggregateTy::Def(aggregate_did, aggregate_args),
```

Comment:
Any way to handle tuples here too?
Even maybe singleton arrays and/or thin pointers while we're at it? (The latter being redundant)